visidata/loaders/mysql.py

```diff
@@ -0,0 +1,114 @@
+from visidata import *
+
+__all__ = ['openurl_mysql', 'MyTable', 'MyTablesSheet']
+
+def codeToType(type_code, colname):
+    import MySQLdb as mysql
+
+    if type_code == mysql.NUMBER:
+        return int
+
+    if type_code == mysql.STRING:
+        return str
+
+    return anytype
+
+
+def openurl_mysql(url, filetype=None):
+    import MySQLdb as mysql
+
+    url = urlparse(url.given)
+    dbname = url.path[1:]
+    conn = mysql.connect(
+                user=url.username,
+                database=dbname,
+                host=url.hostname,
+                port=url.port or 3306,
+                password=url.password,
+                use_unicode=True,
+                charset='utf8')
+
+    return MyTablesSheet(dbname+"_tables", sql=SQL(conn), schema=dbname)
+
+
+class SQL:
+    def __init__(self, conn):
+        self.conn = conn
+
+    def cur(self, qstr):
+        # FIXME: The Postgres loader is using a named cursor but afaik MySQL does not know named cursors… problem?
+        cur = self.conn.cursor()
+        cur.execute(qstr)
+        return cur
+
+    @asyncthread
+    def query_async(self, qstr, callback=None):
+        with self.cur(qstr) as cur:
+            callback(cur)
+            cur.close()
+
+
+def cursorToColumns(cur, sheet):
+    sheet.columns = []
+    for i, coldesc in enumerate(cur.description):
+        name, type, *_ = coldesc
+        sheet.addColumn(ColumnItem(name, i, type=codeToType(type, name)))
+
+
+# rowdef: (table_name, ncols)
+class MyTablesSheet(Sheet):
+    rowtype = 'tables'
+
+    def reload(self):
+        qstr = f'''
+            select
+                t.table_name,
+                column_count.ncols,
+                t.table_rows as est_nrows
+            from
+                information_schema.tables t,
+                (
+                    select
+                        table_name,
+                        count(column_name) as ncols
+                    from
+                        information_schema.columns
+                    where
+                        table_schema = '{self.schema}'
+                    group by
+                        table_name
+                ) as column_count
+            where
+                t.table_name = column_count.table_name;
+        '''
+
+        with self.sql.cur(qstr) as cur:
+            self.nrowsPerTable = {}
```

Comment:
Is this variable used anywhere?

Reply:
Nope, seems to be superfluous already in the Postgres loader…

Reply:
Fixed in cf3966827e005b373de5f3875aab4f056e9cd0a8.